### PR TITLE
feat: add btt tests for exitswap pool amount in

### PR DESCRIPTION
**Pull request by 0xteddybear**

### CODE REVIEW

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.tree [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+│   └── it should revert // subtraction underflow
+├── when computed token amount out is less than minAmountOut
+│   └── it should revert
+├── when computed token amount out exceeds max allowed ratio
+│   └── it should revert
+└── when preconditions are met
+    ├── it emits LOG_CALL event
+    ├── it sets the reentrancy lock
+    ├── it queries token out balance
+    ├── it emits LOG_EXIT event for every token
```
Comment (wei3erHase) [inline test/unit/BPool/BPool_ExitswapPoolAmountIn.tree:16]: not

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.tree [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    ├── it pulls poolAmountIn shares
+    ├── it burns poolAmountIn - exitFee shares
+    ├── it sends exitFee to factory
+    ├── it calls _pushUnderlying for token out
```
Comment (wei3erHase) [inline test/unit/BPool/BPool_ExitswapPoolAmountIn.tree:20]: not!

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+    // it sends exitFee to factory
+    bPool.expectCall__pushPoolShare(deployer, 0);
+    // it calls _pushUnderlying for token out
+    bPool.expectCall__pushUnderlying(tokenOut, address(this), expectedAmountOut);
+    // it emits LOG_CALL event
+    bytes memory _data = abi.encodeCall(IBPool.exitswapPoolAmountIn, (tokenOut, poolAmountIn, expectedAmountOut));
+    vm.expectEmit();
+    emit IBPool.LOG_CALL(IBPool.exitswapPoolAmountIn.selector, address(this), _data);
+    // it emits LOG_EXIT event for every token
+    emit IBPool.LOG_EXIT(address(this), tokenOut, expectedAmountOut);
```
Comment (wei3erHase) [inline test/unit/BPool/BPool_ExitswapPoolAmountIn.t.sol:95]: 👀

**Review by wei3erHase — CHANGES_REQUESTED** [review]
smol typos in the branch definitions, other than that looks ok

File: test/unit/BPool/BPool_ExitswapPoolAmountIn.tree (hunk 2 above)
Comment (0xteddybear) [reply test/unit/BPool/BPool_ExitswapPoolAmountIn.tree:20]: it does :sob: `src/contracts/BPool.sol:434`

**Review by wei3erHase — APPROVED** [review]
looks good now! 🚀

**Pull request merged**